vscode/src/telemetry.ts

```diff
@@ -257,6 +259,26 @@ type EventTypes = {
     };
     measurements: { timeToCompleteMs: number };
   };
+  [EventType.ChatTurnStart]: {
+    properties: {
+      conversationId: string; // associates all events in conversation
+      associationId: string; // associates a start/end events for a single turn
+      serviceUrlHash: string;
+    };
+    measurements: {
+      conversationMessages: number; // includes the one just being submitted
+    };
+  };
+  [EventType.ChatTurnEnd]: {
+    properties: {
+      associationId: string;
+      flowStatus: UserFlowStatus;
+      toolCalls: string; // JSON number array
```

Comment:
I'm not sure what to make of this comment. What is this string going to contain?

Reply:
Yes, wtf was I doing here? I think this was meant to be a list of the _names_ of the tools that were invoked, but maybe I tried to do something clever instead.

`string[]` is not a type we can send, so it's going to be a JSON array that we can easily decode in the query - e.g. `["getJobs", "getWorkspace"]` etc

Reply:
Oh it is a string array, the comment was just wrong. Fixed the comment